autolens/pipeline/phase/abstract/result.py

```diff
@@ -31,6 +33,62 @@ def __init__(
     def most_likely_tracer(self):
         return self.analysis.tracer_for_instance(instance=self.instance)
 
+    @property
+    def source_plane_light_profile_centres(self):
+        if self.most_likely_tracer.source_plane.has_light_profile:
+            return self.most_likely_tracer.light_profile_centres_of_planes[-1]
+        else:
+            return []
```

Comment:
Try and except the error, returning the default in the except. That way you probably don't need the has_light_profile method.

Does the if clause actually return a list?